src/pages/index.js

```diff
@@ -4,42 +4,145 @@ import FormValidator from "../components/FormValidator.js";
+    "Content-Type": "application/json",
+  },
+});
+
+let section;
+let userInfo;
+
+api.userInfoAndCards();
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#cards-template",
+    handleImageClick,
+    handleCardDelete,
+    handleCardLike
+  );
+  return card.generateCard();
 }
+api.getInitialCards().then((data) => {
+  section = new Section({ items: data, renderer: createCard }, ".cards__list");
+  section.renderItems();
+});
+api.getUser().then((data) => {
+  userInfo = new UserInfo(
+    ".profile__title",
+    ".profile__description",
+    ".profile__image"
+  );
+  userInfo.setUserInfo({ title: data.name, description: data.about });
+  userInfo.changeAvatarImage(data.avatar);
+});
 
-function handleProfileFormCreate(inputValues) {
-  const cardElement = createCard(inputValues);
-  section.addItem(cardElement);
-  profileCardPopup.close();
+function handleProfileFormSubmit(formData) {
+  profileSaveButton.textContent = "Saving...";
+  let profileData = { name: newName.value, about: newJob.value };
+  api
+    .editProfile(profileData)
+    .then(() => {
+      userInfo.setUserInfo(formData);
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profileSaveButton.textContent = "Save";
+    });
+}
+function handleProfileFormCreate({ name, link }) {
+  cardCreateSaveButton.textContent = "Saving...";
+  api
+    .addNewCard({ name, link })
+    .then((cardData) => {
+      const cardElement = createCard(cardData);
+      section.addItem(cardElement);
+      profileCardPopup.close();
+      formValidators["card-delete-form"].resetValidation();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      cardCreateSaveButton.textContent = "Save";
+    });
+}
+function handleAvatarFormSubmit({ link }) {
+  avatarSaveButton.textContent = "Saving...";
+  const avatarLoad = { avatar: link };
+  api
+    .updateAvatar(avatarLoad)
+    .then(() => {
+      userInfo.changeAvatarImage(link);
+      avatarEditPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      avatarSaveButton.textContent = "Save";
+    });
 }
 
-function handleImageClick(data) {
-  previewImagePopup.open(data);
+function handleCardDelete(cardData) {
+  cardDeletePopup.open();
+
+  formValidators["card-delete-form"].enableValidation();
+  cardDeletePopup.setConfirmSubmit(() => {
+    const cardId = cardData._id;
+    api
+      .deleteCard({ cardId })
+      .then(() => {
+        cardData.removeCards();
+        cardDeletePopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
 }
 
-export function createCard(data) {
-  const newCard = new Card(data, "#cards-template", handleImageClick);
-  return newCard.generateCard();
+function handleCardLike(cardData) {
+  let cardId = cardData._id;
+  let isLiked = cardData.isLiked;

```

Comment:
If a variable is declared using `let`, it must be **reassigned** somewhere.
If a variable doesn’t change (is not reassigned), it is declared using `const`.

**All your variables** should be declared using `const`.